### DISCUSSION TITLE
S3KeySensor

### DISCUSSION THREAD

**Discussion by sou-joshi** [apache/airflow#16526] [Q&A]
Trying to use S3KeySesor, When it finds a file it continues. But I will want it to return the file name which made it pass to success. Does the sensor return the file name that triggered it to success? or the return value is just a Boolean?

**Comment by josh-fell** [comment]
Hi @sou-joshi,

The `S3KeySensor` will only return a boolean value as it's currently written but you could certainly extend the sensor or the `check_for_wildcard_key()` and/or `check_for_key()` methods in the `S3Hook` to return the key value as needed.  Should you change the sensor's `poke()` method though, just make sure it still returns a boolean so that it functions properly.
